### ISSUE TITLE
[CloudFirestorePlugin] Unchecked Cast

### ISSUE THREAD

**Issue by Vitor-Bukovitz** [FirebaseExtended/flutterfire#2824]
**Describe the bug**
I'm getting 2 warnings of unchecked cast with the CloudFirestorePlugin

```
/home/vitor/flutter/.pub-cache/hosted/pub.dartlang.org/cloud_firestore-0.13.7/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java:309: warning: [unchecked] unchecked cast
        List<Object> values = (List<Object>) value;             
                                             ^                  
  required: List<Object>                                        
  found:    Object                                              
/home/vitor/flutter/.pub-cache/hosted/pub.dartlang.org/cloud_firestore-0.13.7/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java:318: warning: [unchecked] unchecked cast
        List<Object> values = (List<Object>) value;             
                                             ^                  
  required: List<Object>                                        
  found:    Object                                              
2 warnings       
```      

**To Reproduce**
1. In pubspec.yaml: 
`cloud_firestore: ^0.13.7`
2. In android/app/build.gradle:
```
   android {
    compileSdkVersion 29

    lintOptions {
        disable 'InvalidPackage'
    }

    defaultConfig {
        applicationId "<my.package.name>"
        minSdkVersion 23
        targetSdkVersion 29
        multiDexEnabled true
        versionCode 110
        versionName "3.2.5"
        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
        ndk {
            abiFilters 'x86', 'armeabi-v7a','arm64-v8a', 'x86_64'
        }
    }
...
   buildTypes {
       release {
            signingConfig signingConfigs.release
            shrinkResources true
            minifyEnabled true
            useProguard true
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro' 
       }
    }
...
dependencies {
    testImplementation 'junit:junit:4.12'
    androidTestImplementation 'androidx.test:runner:1.2.0'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
    implementation 'com.google.firebase:firebase-analytics:17.2.2'
    implementation 'com.google.firebase:firebase-auth:19.1.0'
    implementation 'com.google.firebase:firebase-messaging:6.0.12'
    implementation 'androidx.multidex:multidex:2.0.1'
}
apply plugin: 'com.google.gms.google-services'
```

**Expected behavior**
Expected no warnings.

**Additional context**
I'm able to build the apk and run the application.

**Flutter doctor**

```
[✓] Flutter (Channel stable, v1.17.4, on Linux, locale en_US.UTF-8)
    • Flutter version 1.17.4 at /home/vitor/flutter
    • Framework revision 1ad9baa8b9 (6 days ago), 2020-06-17 14:41:16 -0700
    • Engine revision ee76268252
    • Dart version 2.8.4

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /home/vitor/Android/Sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /home/vitor/Android/Sdk
    • ANDROID_SDK_ROOT = /home/vitor/Android/Sdk
    • Java binary at: /home/vitor/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Android Studio (version 3.6)
    • Android Studio at /home/vitor/android-studio
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] Connected device (1 available)
    • moto e5 play • ZY3234FQQ9 • android-arm • Android 8.0.0 (API 26)

• No issues found!
```

**Comment by TahaTesser** [comment]
<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel dev, 1.20.0-1.0.pre, on Mac OS X 10.15.5 19F101, locale
    en-GB)
    • Flutter version 1.20.0-1.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision f73f498da1 (8 days ago), 2020-06-18 08:23:22 -0700
    • Engine revision c0d52b98d1
    • Dart version 2.9.0 (build 2.9.0-16.0.dev 2b917f5b6a)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.46.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.11.0

[✓] Connected device (6 available)
    • SM M305F      • 32003c30dc19668f                     • android-arm64  •
      Android 10 (API 29)
    • Taha’s iPhone • 00008020-001059882212002E            • ios            •
      iOS 13.5.1
    • iPhone 11     • 0EC80516-8F99-4A0C-89ED-2273579862F9 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-5 (simulator)
    • macOS         • macOS                                • darwin-x64     •
      Mac OS X 10.15.5 19F101
    • Web Server    • web-server                           • web-javascript •
      Flutter Tools
    • Chrome        • chrome                               • web-javascript •
      Google Chrome 83.0.4103.116

• No issues found!

``` 

</details>

**Comment by Salakar** [comment]
Hey all, as part of our on-going work for #2582, this has been resolved in our Firebase Firestore rework (#2913) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
